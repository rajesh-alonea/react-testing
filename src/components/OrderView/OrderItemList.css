.order-items__quantity {
  width: 36px;
}

.order-items__price {
  width: 110px;
}
