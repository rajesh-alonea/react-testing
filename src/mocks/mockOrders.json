[
  {
    "id": "3f9db14f-9d21-455a-b250-dbe28b05202a",
    "createdAt": "2021-01-01T14:00:00.000Z",
    "items": [
      {
        "productId": "apple-imac",
        "productName": "iMac",
        "price": 1299,
        "quantity": 1
      },
      {
        "productId": "apple-ipad",
        "productName": "iPad",
        "price": 249,
        "quantity": 1
      },
      {
        "productId": "nest-learning-thermostat",
        "productName": "Nest Learning Thermostat",
        "price": 249,
        "quantity": 1
      }
    ],
    "shippingAddress": {
      "firstName": "Elon",
      "lastName": "Musk",
      "company": "Tesla, Inc.",
      "address": "3500 Deer Creek Road",
      "city": "Palo Alto",
      "state": "CA",
      "zip": "94304"
    }
  },
  {
    "id": "68e19edd-8a5e-4af2-a432-2e7ab04e7dcd",
    "createdAt": "2021-07-09T14:00:00.000Z",
    "items": [
      {
        "productId": "apple-imac",
        "productName": "iMac",
        "price": 1299,
        "quantity": 1
      },
      {
        "productId": "apple-ipad",
        "productName": "iPad",
        "price": 249,
        "quantity": 1
      },
      {
        "productId": "nest-learning-thermostat",
        "productName": "Nest Learning Thermostat",
        "price": 249,
        "quantity": 1
      },
      {
        "productId": "gopro-hero",
        "productName": "GoPro HERO Session Action Camera",
        "price": 149,
        "quantity": 4
      }
    ],
    "shippingAddress": {
      "firstName": "Tim",
      "lastName": "Cook",
      "company": "Apple, Inc.",
      "address": "One Apple Park Way",
      "city": "Cupertino",
      "state": "CA",
      "zip": "95014"
    }
  },
  {
    "id": "70871df2-7b8a-40fc-bb33-6d0d9af73dff",
    "createdAt": "2021-03-05T14:00:00.000Z",
    "items": [
      {
        "productId": "canon-eos-rebel",
        "productName": "Canon EOS Rebel SL1",
        "price": 403,
        "quantity": 1
      }
    ],
    "shippingAddress": {
      "firstName": "Mark",
      "lastName": "Zuckerberg",
      "company": "Facebook",
      "address": "1 Hacker Way",
      "city": "Menlo Park",
      "state": "CA",
      "zip": "94025"
    }
  },
  {
    "id": "7df4f6d1-d5a0-454c-9bed-a6284e84c372",
    "createdAt": "2021-02-05T14:00:00.000Z",
    "items": [
      {
        "productId": "fitbit-charge-hr",
        "productName": "Fitbit Charge HR",
        "price": 136,
        "quantity": 1
      }
    ],
    "shippingAddress": {
      "firstName": "Jeff",
      "lastName": "Bezos",
      "company": "Amazon.com, Inc.",
      "address": "2111 7th Ave",
      "city": "Seattle",
      "state": "WA",
      "zip": "98121"
    }
  }
]
