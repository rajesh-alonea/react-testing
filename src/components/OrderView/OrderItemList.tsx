import React from 'react';
import { NumberUtils } from '@react-force/number-utils';
import { OrderItemUtils } from '../../models';
import { OrderItem } from '../../models';
import './OrderItemList.css';

export interface OrderItemListProps {
  items: Array<OrderItem>;
  className?: string;
}

export const OrderItemList = ({ items, className }: OrderItemListProps) => {
  return (
    <table data-testid="order-items" className={className}>
      <tbody>
        {items.map((item, index) => (
          <tr key={index} data-testid="order-item">
            <td>{item.productName}</td>
            <td className="order-items__quantity text-right">
              {item.quantity}
            </td>
            <td
              className="order-items__price text-right"
              data-testid="price-cell"
            >
              {NumberUtils.formatAsMoney(OrderItemUtils.totalItem(item))}
            </td>
          </tr>
        ))}
      </tbody>
      <tfoot>
        <tr>
          <td colSpan={2}>Total</td>
          <td className="text-right">
            {NumberUtils.formatAsMoney(OrderItemUtils.totalItems(items))}
          </td>
        </tr>
      </tfoot>
    </table>
  );
};
