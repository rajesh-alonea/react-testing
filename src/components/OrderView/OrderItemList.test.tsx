import React from 'react';
import mockOrders from '../../mocks/mockOrders.json';
import { render } from '../../test/test-utils';
import { OrderItemList } from './OrderItemList';

describe('<OrderItemList />', () => {
  test('renders correctly', async () => {
    const { asFragment } = render(
      <OrderItemList items={mockOrders[0].items} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
