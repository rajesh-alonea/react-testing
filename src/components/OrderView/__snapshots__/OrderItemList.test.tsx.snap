// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItemList /> renders correctly 1`] = `
<DocumentFragment>
  <table
    data-testid="order-items"
  >
    <tbody>
      <tr
        data-testid="order-item"
      >
        <td>
          iMac
        </td>
        <td
          class="order-items__quantity text-right"
        >
          1
        </td>
        <td
          class="order-items__price text-right"
          data-testid="price-cell"
        >
          1,299.00
        </td>
      </tr>
      <tr
        data-testid="order-item"
      >
        <td>
          iPad
        </td>
        <td
          class="order-items__quantity text-right"
        >
          1
        </td>
        <td
          class="order-items__price text-right"
          data-testid="price-cell"
        >
          249.00
        </td>
      </tr>
      <tr
        data-testid="order-item"
      >
        <td>
          Nest Learning Thermostat
        </td>
        <td
          class="order-items__quantity text-right"
        >
          1
        </td>
        <td
          class="order-items__price text-right"
          data-testid="price-cell"
        >
          249.00
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td
          colspan="2"
        >
          Total
        </td>
        <td
          class="text-right"
        >
          1,797.00
        </td>
      </tr>
    </tfoot>
  </table>
</DocumentFragment>
`;
